### PR TITLE
Wrap functions returning Tensor Lists in Arrays

### PR DESCRIPTION
**Pull request by mib32**
As discussed in #13. Big thanks for the directions! Will be happy to hear feedback and iterate.

This is a contribution from @riboseinc.

### CODE REVIEW

File: ext/torch/templates.cpp [hunk 1]
```diff
@@ -53,3 +54,10 @@ Object wrap(std::tuple<torch::Tensor, torch::Tensor, double, int64_t> x) {
   a.push(to_ruby<int64_t>(std::get<3>(x)));
   return Object(a);
 }
+
+Array wrap(std::vector<torch::Tensor> x) {
```
Comment (ankane) [inline ext/torch/templates.cpp:58]: Let's return an `Object` to be consistent with the other `wrap` methods.

File: ext/torch/templates.cpp [hunk 2]
```diff
@@ -53,3 +54,10 @@ Object wrap(std::tuple<torch::Tensor, torch::Tensor, double, int64_t> x) {
...
+  Array a;
+  for(torch::Tensor t : x)
```
Comment (ankane) [inline ext/torch/templates.cpp:60]: `torch::Tensor& t` or `auto& t` should avoid copying the tensor.

File: test/tensor_methods_test.rb [hunk 3]
```diff
@@ -143,4 +143,28 @@ def test_fill_diagonal
     assert_equal [[3, 1], [1, 3]], x.fill_diagonal!(3).to_a
     assert_equal [[3, 1], [1, 3]], x.to_a
   end
+
+  def test_chunk
+    x = Torch.ones(2, 2)
+    assert_equal [[[1, 1], [1, 1]]], x.chunk(1, 1).to_a.map(&:to_a)
```
Comment (ankane) [inline test/tensor_methods_test.rb:149]: Small one: `to_a.map(&:to_a)` can be simplified to `map(&:to_a)`.

File: test/tensor_methods_test.rb [hunk 4]
```diff
@@ -143,4 +143,28 @@ def test_fill_diagonal
...
+    assert_equal [[[1], [1]], [[1], [1]]], x.chunk(2, 1).to_a.map(&:to_a)
+  end
+
+  def test_split
```
Comment (ankane) [inline test/tensor_methods_test.rb:153]: Thanks for testing this as well

**Review by ankane — COMMENTED** [review]
Thanks @mib32 (and @riboseinc), looks really great! Left a few small comments inline.

**Comment by mib32** [comment]
Thanks for the valuable feedback. Hope it looks better now. 
Also added a test for #unbind, because it's the other method that became wrapped in this PR.

**Comment by ankane** [comment]
Awesome, looks really great. Thanks @mib32!

**Comment by ankane** [comment]
Hey @mib32, just fyi, I recently profiled the library and found a significant amount of time was being spent deciding which LibTorch C++ function to call from Ruby. That logic's been moved to C++ (like PyTorch) on master, so you may want to see if it speeds up your training.

A rough micro benchmark calling `add` 1,000,000 times:

Version | Time (sec) | Note
--- | --- | ---
0.3.6 | 20.3 |
0.3.7 | 12.3 | Ruby optimizations, released today
master | 4.4 | C++ version

**Comment by mib32** [comment]
@ankane Aha! Yes, I will update right now. That's so very kind of you to notify 🌟

**Pull request merged**